source/extensions/tracers/xray/reservoir.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <chrono>
+
+#include "envoy/common/time.h"
+
+#include "common/common/lock_guard.h"
+#include "common/common/thread.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+/**
+ * Simple token-bucket algorithm that enables counting samples/traces used per second.
+ */
+class Reservoir {
+public:
+  /**
+   * Creates a new reservoir that allows up to |traces_per_second| samples.
+   */
+  explicit Reservoir(unsigned traces_per_second)
```

Comment:
last one: uint32_t

Reply:
ok. I did a grep this time and it is indeed the last one. Sorry about that.